SmartTutor/Recommenders/KnowledgeBasedRecommender.cs

```diff
@@ -20,5 +27,39 @@ public List<LearningObject> FindEducationalContent(List<SmellType> issues)
             //TODO: Contact ContentService 
             return result;
         }
+
+        public NodeProgress BuildNodeProgressForTrainee(Trainee trainee, KnowledgeNode knowledgeNode)
+        {
+            var learningObjects = new List<LearningObject>();
+            var sortedPreferences = trainee.LearningPreferenceScore().OrderByDescending(key => key.Value).ToList();
+
+            foreach (var summary in knowledgeNode.LearningObjectSummaries)
+            {
+                foreach (var learningObject in sortedPreferences
+                    .Select(preference => GetLearningObjectForPreference(preference.Key, summary.Id))
+                    .Where(learningObject => learningObject != null))
```

Comment:
We should return one LO for each summary for now. As far as I can tell this code also won't return anything if there is only a challenge tied to a summary (or new types we will add in the future). We should always have a fallback - if you don't find anything, select the first available.